src/v/cluster/partition.cc

```diff
@@ -894,7 +896,7 @@ partition::transfer_leadership(std::optional<model::node_id> target) {
         vlog(clusterlog.trace, "transfer_leadership[{}]: no archiver", ntp());
     }
 
-    co_return co_await _raft->do_transfer_leadership(target);
+    co_return co_await _raft->do_transfer_leadership(req);
```

Comment:
Is it okay that the  25ms timeout will not apply to the tiered  storage graceful shutdown above?  It is still bounded by `cloud_storage_graceful_transfer_timeout_ms`  (5 seconds).

It's basically instant for typical partitions that are only occasionally mid-upload, but it will be slower (~1-2 seconds typically) for high throughput partitions that have a good chance of being in the middle of an upload at any given moment.

Reply:
Nice find. As we discussed on Slack I'll move the `prepare_transfer_leadership` for the archiver to above where we get the `stm_prepare_lock` for the `rm/tm_stm`s. This will allow the archiver to take the time it needs to cleanly shutdown and avoid having append requests to the partition be blocked for longer than the transfer request's timeout.